core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableSchemaBuilder.java

```diff
@@ -114,12 +115,12 @@ public void setSortColumns(List<ColumnSchema> sortColumns) {
     this.sortColumns = sortColumns;
   }
 
-  public ColumnSchema addColumn(StructField field, boolean isSortColumn) {
-    return addColumn(field, null, isSortColumn, false);
+  public ColumnSchema addColumn(StructField field, AtomicInteger valIndex, boolean isSortColumn) {
```

Comment:
Do we need Atomic Integer ? This is single threaded. Later phase also even if writer becomes multi threaded, building schema and CarbonTable should remain single threaded only.

Reply:
AtomicInteger is not used here for multi threading purpose. It is used for incrementing a value for complex boolean array type to assign child column name